bpf/dns_tracker.h

```diff
@@ -108,6 +108,14 @@ static __always_inline int track_dns_packet(struct __sk_buff *skb, pkt_info *pkt
             }
             pkt->dns_id = dns_id;
             pkt->dns_flags = flags;
+
+            // Copy raw QNAME bytes (label-encoded) and let userspace decode to dotted form
+            __builtin_memset(pkt->dns_name, 0, DNS_NAME_MAX_LEN);
+            u32 qname_off = dns_offset + sizeof(struct dns_header);
+            // Best-effort fixed-size copy; safe for verifier (constant size)
+            (void)bpf_skb_load_bytes(skb, qname_off, pkt->dns_name, DNS_NAME_MAX_LEN - 1);
+            // Ensure null-termination
+            pkt->dns_name[DNS_NAME_MAX_LEN - 1] = '\0';
```

Comment:
I wonder if we should move that outside of the query / response condition.

Question header can be provided in both cases and that may help getting more results but on the other hand that copy function will be called more often.

Reply:
I'm investigating that way but indeed it will consume more resources and the output is not convincing to me.
I guess we can keep things as is here.